src/sas/qtgui/Plotting/Plotter.py

```diff
@@ -250,7 +252,7 @@ def plot(self, data=None, color=None, marker=None, hide_error=False, transform=T
         self.plot_lines[data.name] = line
 
         # Now add the legend with some customizations.
-        if self.showLegend:
+        if self.showLegend and self.legendVisible:
```

Comment:
This is suppressing legend creation on redraw so the legend disappears with no way to get it back. Steps to repeat:
- Plot data
- Toggle legend to make it invisible
- Change the scale to force a redraw
- Try (and fail) to toggle the legend to make it visible again

Suggestion:
```
if self.showLegend:
    ...  # after legend is created
    self.legend.set_visible(self.legendVisible)
```

Reply:
Fixed, it can now actually toggle, thanks!